components/ReplySearch/ReplySearch.js

```diff
@@ -0,0 +1,170 @@
+            article={article}
+            onConnect={onConnect}
+          />
+        );
+      })}
+      <style jsx>{`
+        .items {
+          list-style-type: none;
+          padding-left: 0;
+        }
+      `}</style>
+    </ul>
+  );
+};
+
+export default class ReplySearch extends PureComponent {
+  state = {
+    tab: false, // reply || article
+    search: '',
+  };
+
+  handleSearch = event => {
+    const { target: { value }, key } = event;
+
+    if (key === 'Enter') {
+      this.setState({ search: value });
+      this.props.onSearch(event);
+    }
+  };
+
+  handleTabChange = tab => () => {
+    this.setState({ tab });
+  };
+
+  renderTabMenu = () => {
+    const { articles, replies } = this.props;
+    const { tab } = this.state;
+    const replisCount = replies.size;
+    const articlesCount = articles.size;
+
+    return (
+      <ul className="tabs">
+        <li
+          onClick={this.handleTabChange('reply')}
+          className={`tab ${tab === 'reply' ? 'active' : ''} ${
+            replisCount === 0 ? 'disabled' : ''
+          }`}
+        >
+          {replisCount === 0 ? (
+            '查無相關文章'
+          ) : (
+            <span>
+              使用相關回應 <span className="badge">{replisCount}</span>
+            </span>
+          )}
+        </li>
+        <li
+          onClick={this.handleTabChange('article')}
+          className={`tab ${tab === 'article' ? 'active' : ''} ${
+            articlesCount === 0 ? 'disabled' : ''
+          }`}
+        >
+          {articlesCount === 0 ? (
+            '查無相關文章'
+          ) : (
+            <span>
+              瀏覽相關文章 <span className="badge">{articlesCount}</span>
+            </span>
+          )}
+        </li>
+        <li className="empty" />
+        <style jsx>{`
+          .tabs {
+            margin-top: 20px;
+          }
+        `}</style>
+        <style jsx>{tabMenuStyle}</style>
+      </ul>
+    );
+  };
+
+  renderSearchReplyTab = () => {
+    const { articles, replies, onConnect } = this.props;
+    const { tab } = this.state;
+
+    const getArticleSimilarity = relatedArticleText =>
+      stringSimilarity.compareTwoStrings(this.search, relatedArticleText);
+
+    switch (tab) {
+      case 'reply':
+        return (
+          <RelatedReplies
+            onConnect={onConnect}
+            relatedReplies={replies}
+            getArticleSimilarity={getArticleSimilarity}
+          />
+        );
+
+      case 'article':
+        return (
+          <SearchArticles onConnect={onConnect} searchArticles={articles} />
+        );
+
+      default:
+        return null;
+    }
+  };
+
+  render() {
+    const { search } = this.state;
+    const { articles, replies } = this.props;
+
+    return (
+      <div>
+        <label htmlFor="replySeach">
+          搜尋相關回應：
+          <input id="replySeach" type="search" onKeyUp={this.handleSearch} />
+        </label>
+
+        {articles.size || replies.size
+          ? [
```

Comment:
This array here introduces key errors:

<img width="1232" alt="2018-02-08 1 02 57" src="https://user-images.githubusercontent.com/108608/35930175-34ccab0a-0c6c-11e8-8730-71638217d053.png">


Please use `<Fragment>` instead.

Although we did not upgrade babel to support `<>` and `</>`, we can still import `<Fragment>`:

```
import React, {Fragment} from React;
```